helix-core/src/selection.rs

```diff
@@ -566,9 +568,9 @@ impl Selection {
     }
 
     /// Takes a closure and maps each `Range` over the closure.
-    pub fn transform<F>(mut self, f: F) -> Self
+    pub fn transform<F>(mut self, mut f: F) -> Self
     where
-        F: Fn(Range) -> Range,
+        F: FnMut(Range) -> Range,
```

Comment:
Where is this FnMut used?

I was actually looking at making the same change for the tree-sitter TreeCursor across injection layers 😀 https://github.com/the-mikedavis/helix/commit/eee9af9f339ebe2aacf9133a9613257d7c2d7bf4#diff-f218cebd639a867087e2a7abc84160d45888ad5bcff4ab10fcf7217cb16586e9L544-R546

Reply:
I needed it because vertical movements remove virtual text lines from the `text_annotations` struct. If there is a virtual text line and we press `j` we want the cursor to got below the virtual text line. If the virtual text line was not removed it would stay in place.. To accomplish this the `clear_virtual_lines` function is called in the movement function. I could avoid the `FnMut` here by simply clearing the `text_annotation` before passing them to the movement or cloning them. But I want to avoid cloning for performance reasons and I don't want to make the caller responsible for clearing the `text_annotations` because then it would be way too easy to accidentally use the movement functions wrong in helix core.

As a sidenote: I had to change `Fn` to `FnMut` a few times now. Whenever I design APIs I always make closures `FnMut` instead of `Fn` unless it requires extra effort to offer maximum flexibility and avoid having to change it in the future